pkg/webhook/v1alpha3/pod/inject_webhook.go

```diff
@@ -143,7 +143,6 @@ func (s *sidecarInjector) Mutate(pod *v1.Pod, namespace string) (*v1.Pod, error)
 	}
 	mutatedPod.Spec.Containers = append(mutatedPod.Spec.Containers, *injectContainer)
 
-	mutatedPod.Spec.ServiceAccountName = pod.Spec.ServiceAccountName
```

Comment:
I had questioned it https://github.com/kubeflow/katib/pull/716#discussion_r317462741, but never go further. By test, it proves this line can be deleted